codec-http2/src/test/java/io/netty/handler/codec/http2/DefaultHttp2ConnectionDecoderTest.java

```diff
@@ -366,16 +374,18 @@ public void emptyDataFrameShouldApplyFlowControl() throws Exception {
         }
     }
 
-    @Test(expected = Http2Exception.class)
+    @Test
     public void dataReadForStreamInInvalidStateShouldThrow() throws Exception {
         // Throw an exception when checking stream state.
         when(stream.state()).thenReturn(Http2Stream.State.CLOSED);
         final ByteBuf data = dummyData();
-        try {
-            decode().onDataRead(ctx, STREAM_ID, data, 10, true);
-        } finally {
-            data.release();
-        }
+        assertThrows(Http2Exception.class, new Executable() {
+            @Override
+            public void execute() throws Throwable {
+                decode().onDataRead(ctx, STREAM_ID, data, 10, true);
+            }
+        });
+        data.release();
```

Comment:
Note that if the assertion fails, this pattern will leak. Not sure how much we care about buffer leaks in _failing_ tests, though.

Reply:
yeah... I think it is fine either way.